### ISSUE TITLE
Next Greater Element [Java]

### ISSUE THREAD

**Issue by SauravP97** [iiitv/algos#672]
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS).

- [ ] Issue Title is in form `Algo/DS Name [Language]`.
- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This Algo/DS is worth adding to this repository.
- [ ] Assign this issue to me (I want to work on this).
	- [ ] I ensure that I am not already assigned to 2 or more issues.
- [ ] I will send my PR only after I'm assigned to this issue by any maintainer.
- [ ] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

**Comment by kavyatripathi** [comment]
@iiitv/project-maintainers 
Can you please explain the problem statement a little bit ?

**Comment by kavyatripathi** [comment]
Is it like for every element in an array find the next greater element ?

**Comment by AmanRaj1608** [comment]
Yeah like giver an array [1,2,3,4,5] you have to tell every element which is greater than that value

For Ex - 
1 -> 2
2 -> 3
3 -> 4
4 -> 5
5 -> -1

**Comment by AmanRaj1608** [comment]
I think this is a general problem, not an algorithm which should be added

others opinion @iiitv/project-maintainers  ?

**Comment by kavyatripathi** [comment]
@iiitv/project-maintainers i would like to work on this issue

**Comment by kavyatripathi** [comment]
@iiitv/project-maintainers  I have created a PR. Please take a look.
